src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/GeneratedDocumentPublisher.cs

```diff
@@ -117,21 +119,23 @@ public void PublishHtml(ProjectKey projectKey, string filePath, SourceText sourc
                 previouslyPublishedData = PublishData.Default;
             }
 
+            if (previouslyPublishedData.HostDocumentVersion > hostDocumentVersion)
+            {
+                // We've already published a newer version of this document. No-op.
+                Debug.Fail("Html document being published that is older than one we've previously published!");
```

Comment:
@ryzngard reports that removing this line (and the equiv for C#) fixes issues in VS Code, so this bears removal, and subsequent investigation